src/planner/World.cpp

```diff
@@ -151,5 +151,50 @@ std::mutex& World::getMutex() const
   return mMutex;
 }
 
+//==============================================================================
+bool World::State::equals(const State& other)
+{
+  if (configurations.size() != other.configurations.size())
+    return false;
+
+  for (size_t idx = 0; idx < configurations.size(); ++idx)
+  {
+    if (configurations[idx] != other.configurations[idx])
+      return false;
+  }
+
+  return true;
+}
```

Comment:
:neckbeard: I think this whole function could be replaced with:
```c++
return configurations == other.configurations;
``` 

Something is also a bit funny about this. Shouldn't we verify that the two `Configuration`s contain the same `Skeleton`s somehow?

Reply:
We're not checking whether the skeletons are the same because the skeletons may be cloned ones. Should I check whether they have the same joint structure?